google-startup-scripts/usr/lib/systemd/system/google-shutdown-scripts.service

```diff
@@ -0,0 +1,15 @@
+[Unit]
+Description=Google Compute Engine user shutdown scripts.
+After=local-fs.target network-online.target network.target
+After=google.service rsyslog.service
```

Comment:
Also you need:
Before=shutdown.target reboot.target halt.targe
To make this run on shutdown/reboot events.